packages/block-editor/src/hooks/utils.js

```diff
@@ -35,11 +35,10 @@ export const cleanEmptyObject = ( object ) => {
  * Supports strings, numbers and arrays:
  *
  * 'foo' => [ 'foo' ]
- * 'foo.bar.baz' => [ 'foo', 'bar', 'baz' ]
  * 2 => [ '2' ]
  * [ 'foo', 'bar' ] => [ 'foo', 'bar' ]
  *
- * @param {number|string|Array} path Path
+ * @param {string|number|Array} path Path
```

Comment:
I'd go a step further and enforce it as an array, always, but this is probably a good middle-ground between convenience and simplicity.

Reply:
Yeah, I think that could be a good improvement for the future, but I decided to leave that support since we have at least one existing use case for a single string path: 

https://github.com/WordPress/gutenberg/blob/d84b05ab8f565c1103c0da7aa087f47a895becb7/packages/block-editor/src/components/editor-styles/index.js#L71